onnxruntime/test/providers/cpu/reduction/reduction_ops_test.cc

```diff
@@ -2166,6 +2175,41 @@ TEST(ReductionOpTest, ArgMax2D_dim1) {
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});
 }
 
+// CoreML EP currently handles a special case for supporting ArgMax op:
+// An ArgMax followed by a Cast to int32 type. 
+// Please see in <repo_root>/onnxruntime/core/providers/coreml/builders/impl/argmax_op_builder.cc
+// and /cast_op_builder.cc. We have the following UT test here for this special case
+// This test case can also be shared later if we want to support similar cases in NNAPI
+#ifdef USE_COREML
```

Comment:
Let's remove CoreML EP dependency on this, make it also a working case for CPU

Reply:
Sure. but this test case uses `TestCoreMLExecutionProvider` which only works when use_coreml, how should I change it?